usr/src/uts/common/fs/zfs/dmu.c

```diff
@@ -739,9 +758,29 @@ dmu_free_long_range_impl(objset_t *os, dnode_t *dn, uint64_t offset,
 		ASSERT3U(chunk_begin, >=, offset);
 		ASSERT3U(chunk_begin, <=, chunk_end);
 
-		dmu_tx_t *tx = dmu_tx_create(os);
-		dmu_tx_hold_free(tx, dn->dn_object,
-		    chunk_begin, chunk_end - chunk_begin);
+		chunk_len = chunk_end - chunk_begin;
+
+		mutex_enter(&dp->dp_lock);
+		for (int t = 0; t < TXG_SIZE; t++) {
+			long_free_dirty_all_txgs +=
+			    dp->dp_long_free_dirty_pertxg[t];
+		}
+		mutex_exit(&dp->dp_lock);
+		free_dirty_pct = 100 * long_free_dirty_all_txgs /
```

Comment:
I think it would be better to calculate `dirty_frees_threshold` in bytes rather than as a percentage. This way we do not need to do any divisions in the loop and can compare `long_free_dirty_all_txgs` directly with `dirty_frees_threshold`.